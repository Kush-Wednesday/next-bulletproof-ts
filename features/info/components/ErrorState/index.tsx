import React from "react";
import styled from "styled-components";
import { Card } from "antd";
import { T } from "@common";

const Error = styled.div`
  color: "red";
`;

const ErrorState = () => {
  return (
    <Error>
      <Card>
        <T text="Something went wrong" />
      </Card>
    </Error>
  );
};

export default ErrorState;
