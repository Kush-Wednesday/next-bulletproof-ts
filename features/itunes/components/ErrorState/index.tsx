/**
 *
 * ErrorState
 *
 */

 import React from "react";
 import get from "lodash/get";
 import { compose } from "redux";
 import { injectIntl, IntlShape } from "react-intl";
 import { T, CustomCard } from "../../../../common";
 import { SongResponse } from "@features/itunes/api/getSongs";
 
 interface ErrorStateProps {
   intl: IntlShape;
   loading: boolean;
   trackData: SongResponse | undefined;
   tracksError: string | undefined;
 }
 
 const ErrorState: React.FC<ErrorStateProps> = ({ intl, trackData, tracksError, loading }) => {
   let trackError: string | undefined;
   if (tracksError) {
    trackError = tracksError;
   } else if (!get(trackData, "resultCount", 0)) {
    trackError = "";
   }
 
   return !loading && trackError ? (
     <CustomCard
       color={tracksError ? "red" : "grey"}
       data-testid="error-state"
     >
       <T id={trackError} />
     </CustomCard>
   ) : null;
 };
 
 export default compose(injectIntl)(ErrorState);
 