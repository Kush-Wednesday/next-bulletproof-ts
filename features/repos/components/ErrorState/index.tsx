/**
 *
 * ErrorState
 *
 */

import React from "react";
import get from "lodash/get";
import { compose } from "redux";
import PropTypes from "prop-types";
import { injectIntl } from "react-intl";
import { T, CustomCard } from "@common";
import { commonPropTypes } from "@utils";

const ErrorState = props => {
  const { intl, reposError, loading, reposData } = props;
  let repoError;
  if (reposError) {
    repoError = reposError;
  } else if (!get(reposData, "totalCount", 0)) {
    repoError = "respo_search_default";
  }
  if (!loading && !repoError) {
    return null;
  }
  return (
    !loading &&
    repoError && (
      <CustomCard
        color={reposError ? "red" : "grey"}
        title={intl.formatMessage({ id: "repo_list" })}
        data-testid="error-state"
      >
        <T id={repoError} />
      </CustomCard>
    )
  );
};
const types = {
  reposData: PropTypes.arrayOf(
    PropTypes.shape({
      totalCount: PropTypes.number,
      incompleteResults: PropTypes.bool,
      items: PropTypes.array,
    })
  ),
  reposError: PropTypes.object,
  repoName: PropTypes.string,
  recommendations: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.number.isRequired, name: PropTypes.string.isRequired })
  ),
};

const { intl, loading } = commonPropTypes;
const { reposError, reposData } = types;

ErrorState.propTypes = {
  intl,
  loading,
  reposError,
  reposData,
};

export default compose(injectIntl)(ErrorState);
