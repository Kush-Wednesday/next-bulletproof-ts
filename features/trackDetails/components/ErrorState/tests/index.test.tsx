import { fireEvent, render } from "@utils/testUtils";
import ErrorState from "../index";

describe("<ErrorState />", () => {
  it("should match the snapshot", () => {
    const { baseElement } = render(<ErrorState />);
    expect(baseElement).toMatchSnapshot();
  });
});
