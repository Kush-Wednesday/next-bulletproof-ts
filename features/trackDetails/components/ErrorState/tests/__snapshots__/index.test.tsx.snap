// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorState /> should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="styled__AlignCenter-sc-15gkald-3 tOKel"
    >
      <div
        class="ErrorState__Error-sc-14nnqu4-0 iiAFPa"
      >
        Sorry. Something went wrong! Please try again in sometime.
      </div>
    </div>
  </div>
</body>
`;
